### PR TITLE
7832 - Adjusted position and hover color for custom button

### LINKED ISSUE #7832: Searchfield: icon hover color have different size
**Describe the bug**
x icon hover color size is different from custom icon

**To Reproduce**

1. Go to https://main-enterprise.demo.design.infor.com/components/searchfield/example-buttons.html
2. Type a text in searchfield
3. hover on each icon

**Expected behavior**

**Version**

- ids-enterprise: v4.87.0-dev
- 

**Screenshots**
image
vs
image

**Platform**
All major browsers

**Additional context**

### PR DESCRIPTION
**Pull request by yohannahbautista**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Adjusted position and hover color for custom button

**Related github/jira issue (required)**:

Closes: 
- #7832 
- #7911 

**Steps necessary to review your pull request (required)**:

- Pull this branch, build and run the app
- Go to: http://localhost:4000/components/searchfield/example-buttons.html
- Hover on barcode, hover should not be so big
- Type in searchfield, hover on barcode, hoves should stay the same

Also check the following in mobile:
- http://localhost:4000/components/toolbar-flex/example-search-before-buttonset.html
- http://localhost:4000/components/toolbar-flex/example-searchfield-collapsible.html
- http://localhost:4000/components/toolbar-flex/example-toolbar-flex-with-searchfield-in-center.html

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -61,6 +61,7 @@
 - `[Popover]` Fixed where popover connected on click will not close on click (it just reopened). ([#7679](https://github.com/infor-design/enterprise/issues/7679))
 - `[Sparkline]` Fixed median fill on dark theme. ([#7717](https://github.com/infor-design/enterprise/issues/7717))
 - `[Searchfield]` Adjusted height for go button. ([#6695](https://github.com/infor-design/enterprise/issues/6695))
+- `[Searchfield]` Adjusted position and hover color for custom button. ([#7832](https://github.com/infor-design/enterprise/issues/7832))
```
Comment (EdwardCoyle) [inline docs/CHANGELOG.md:64]: Very minor but I think this belongs under v4.88

**Review by EdwardCoyle — APPROVED** [review]
Feature looks great!  Just a note below:

File: docs/CHANGELOG.md (hunk 1 above)
Comment (tmcconechy) [reply docs/CHANGELOG.md:64]: Yes on 4.88 now

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Seems to be an issue with the component on mobile view ports.

Screenshot 2023-09-06 at 1 27 29 PM

a) the buttons are for from the right (see b)
b) the left side icon to collapse is on the wrong side
c) if you expand and then collapse it looks strange (buttons all out of place)

**Review by clayinfor — DISMISSED** [review]

**Review by tjamesallen15 — DISMISSED** [review]

**Comment by tmcconechy** [comment]
It looks a little funny when you expand and then collapse it 

1) Leaves it in wrong place
Screenshot 2023-09-11 at 10 52 04 AM
2) can animation be improved?

**Review by tmcconechy — APPROVED** [review]

**Review by janahintal — CHANGES_REQUESTED** [review]
Hello, noticed the following upon testing. 

No ellipsis when not hovered or focus
image

Focus overlaps in ellipsis in responsive view. 
image

Too much space on bar code and x icon
image

**Comment by jbrcna** [comment]
the ellipsis is shown even if there still a space left for characters.
key in "the quick brown fox"
image

in classic, bar code icon is not displayed.
image

**Review by glenlieorillo — CHANGES_REQUESTED** [review]
Tested also in BrowserStack.

https://7832-searchfield-custom-button-enterprise.demo.design.infor.com/components/searchfield/example-buttons.html

**1. The text and barcode are overlapping.**
image

**2. The issue found by @tmcconechy still exists on mobile.**

Steps to reproduce:
1. Type in any text in search bar
2. Click close button
3. See error

image

**3. Barcode is not aligned properly.**
image

**4. Too much space between barcode and close button.**
image

**Review by tmcconechy — DISMISSED** [review]
@janahintal @glenlieorillo anything else you think we need to fix here?
Note this is a minor pattern. And we should focus on the original issue

**Review by ericangeles — APPROVED** [review]

**Review by glenlieorillo — CHANGES_REQUESTED** [review]
'Sort' button is cut-off and the 'Sort' option also appears on the dropdown menu.

https://7832-searchfield-custom-button-enterprise.demo.design.infor.com/components/toolbar-flex/example-search-before-buttonset.html

image

Menu button is cut-off and it also appears on the dropdown menu.

https://7832-searchfield-custom-button-enterprise.demo.design.infor.com/components/toolbar-flex/example-searchfield-collapsible.html

image

**Review by jbrcna — CHANGES_REQUESTED** [review]
in classic, bar code icon is not displayed.
image

icon not centered
image
image

**Comment by tmcconechy** [comment]
@yohannahbautista the issue with the button cut off is not fixable. This is how the toolbar works. It may partly cut off the button depending on the width but perhaps @yohannahbautista can look a bit at the other part(s)

**Comment by ericangeles** [comment]
@jbrcna could you create a separate ticket for the second one?

**Review by tmcconechy — DISMISSED** [review]

**Review by tmcconechy — APPROVED** [review]
Is this all done aside from future issues? I think its good enough as is. Wierd pattern not used much

**Comment by tmcconechy** [comment]
@ericangeles what should we do this this one? I think maybe merge and raise new issues for anything related.

**Comment by ericangeles** [comment]
@tmcconechy, yeah, let's merge this PR and create a separate issue(s) if there are any.

**Review by tmcconechy — APPROVED** [review]

**Pull request merged**